great_expectations/datasource/new_datasource.py

```diff
@@ -261,6 +261,45 @@ def get_available_data_asset_names(
 
         return available_data_asset_names
 
+    def get_available_data_asset_names_and_types(
+        self, data_connector_names: Optional[Union[list, str]] = None
+    ) -> Dict[str, List[Tuple[str, str]]]:
+        """
+        Returns a dictionary of data_asset_names that the specified data
+        connector can provide. Note that some data_connectors may not be
+        capable of describing specific named data assets, and some (such as
+        inferred_asset_data_connector) require the user to configure
+        data asset names.
+
+        Returns:
+            dictionary consisting of sets of data assets available for the specified data connectors:
+            For instance, in a SQL Database the data asset name corresponds to the table or
+            view name, and the data asset type is either 'table' or 'view'.
```

Comment:
If the data asset type is one of two values, should we maybe add an assertion?

Reply:
hmm, I'm open to this, but not yet seeing the value added. I'm not sure that BaseDatasource has a great interest in exactly what the type is, just that it exists. That comment is there to help clarify the choice of/give some context to the variable name `type`, which tends to be pretty overloaded.